bread/bread.py

```diff
@@ -204,6 +204,9 @@ def get_queryset(self):
 
     def get_context_data(self, **kwargs):
         data = super(BrowseView, self).get_context_data(**kwargs)
+        q = self.request.GET.get('q', False)
+        if q:
+            data['q'] = q
```

Comment:
It seems a bit brittle to me that `q` is only conditionally in the context data when the template references it unconditionally. I know that Django's default template resolution for undefined context variables is to emit '', so this works for now, but it just seems like a pitfall for some future developer (e.g. someone who edits the template, sees `q`, and doesn't bother to check whether or not its conditionally present).

Besides, `data['q'] = q if q else ''` is fewer lines =)